docs/styleguide.md

```diff
@@ -250,10 +250,80 @@ The properties chapter must be in following form:
 The heading can be `###` or `####`-levels depending on whether the property
 belongs to a module or a class.
 
+## API History
+
+An "API History" block is a YAML code block encapsulated by a HTML comment that
+should be placed directly after the Markdown header for a class or method, like so:
+
+`````markdown
+#### `win.setTrafficLightPosition(position)` _macOS_
+
+<!--
+```YAML history
+added:
+  - pr-url: https://github.com/electron/electron/pull/22533
+changes:
+  - pr-url: https://github.com/electron/electron/pull/26789
+    description: Made `trafficLightPosition` option work for `customButtonOnHover` window.
+deprecated:
+  - pr-url: https://github.com/electron/electron/pull/37094
+    breaking-changes-header: deprecated-browserwindowsettrafficlightpositionposition
+```
+-->
+
+* `position` [Point](structures/point.md)
+
+Set a custom position for the traffic light buttons. Can only be used with `titleBarStyle` set to `hidden`.
+`````
+
+It should adhere to the API History [JSON Schema](https://json-schema.org/)
+(`api-history.schema.json`) which you can find in the `docs` folder.
+The [API History Schema RFC][api-history-schema-rfc] includes example usage and detailed
+explanations for each aspect of the schema.
+
+The purpose of the API History block is to describe when/where/how/why an API was:
+
+* Added
+* Changed (usually breaking changes)
+* Deprecated
+
+Each API change listed in the block should include a link to the
+PR where that change was made along with an optional short description of the
+change. If applicable, include the [heading id](https://gist.github.com/asabaylus/3071099)
+for that change from the [breaking changes documentation](./breaking-changes.md).
+
+The [API History linting script][api-history-linting-script] (`lint:api-history`)
+validates API History blocks in the Electron documentation against the schema and
+performs some other checks. You can look at its [tests][api-history-tests] for more
+details.
+
+There are a few style guidelines that aren't covered by the linting script:
+
+* Always adhere to this format:
+
+  | Type                        | Line                              |
+  |----------------------------:|:----------------------------------|
+  | `API HEADER`                | `` #### `win.flashFrame(flag)` `` |
+  | `BLANK LINE`                | `​`                                |
+  | `HTML COMMENT OPENING TAG`  | `` <!-- ``                        |
+  | `API HISTORY OPENING TAG`   | `` ```YAML history ``             |
+  | `API HISTORY`               | `added: ...`                      |
+  | `API HISTORY CLOSING TAG`   | `` ``` ``                         |
+  | `HTML COMMENT CLOSING TAG`  | `` --> ``                         |
+  | `BLANK LINE`                | `​`                                |
```

Comment:
IMO this is a little confusing if you don't know the format already. I think there might be a better way to convey the info.

Reply:
Yeah I agree, I just couldn't think of anything better 😅. I'll try out some other ideas.

Reply:
Do you like any of these? @dsanders11 @erickzhao @VerteDinde

<details>
<summary>

### HTML Comment Left-Align (1)

</summary>

  `````markdown
  <!-- API HEADER -->                   |  #### `win.flashFrame(flag)`
  <!-- BLANK LINE -->                   |
  <!-- HTML COMMENT OPENING TAG -->     |  <!--
  <!- - API HISTORY OPENING TAG - ->    |  ```YAML history
  <!- - API HISTORY - ->                |  added:
                                        |    - pr-url: https://github.com/electron/electron/pull/22533
  <!- - API HISTORY CLOSING TAG - ->    |  ```
  <!- - HTML COMMENT CLOSING TAG - ->   |  -->
  <!-- BLANK LINE -->                   |
  `````

</details>

<details>
<summary>

### HTML Comment Right-Align (2)

</summary>

`````markdown
  #### `win.flashFrame(flag)`   <!-- API HEADER -->
                                <!-- BLANK LINE -->
  <!--                          <!- - HTML COMMENT OPENING TAG -->
  ```YAML history               <!- - API HISTORY OPENING TAG -->
  added:                        <!- - API HISTORY -->
    - pr-url: https://...
  ```                           <!- - API HISTORY CLOSING TAG -->
  -->                           <!-- HTML COMMENT CLOSING TAG -->
                                <!-- BLANK LINE -->
  `````

</details>

<details>
<summary>

### Plaintext Comment Left-Align (3)

</summary>

  `````markdown
  API HEADER                  |  #### `win.flashFrame(flag)`
  BLANK LINE                  | 
  HTML COMMENT OPENING TAG    |  <!--
  API HISTORY OPENING TAG     |  ```YAML history
  API HISTORY                 |  added:
                              |    - pr-url: https://github.com/electron/electron/pull/22533
  API HISTORY CLOSING TAG     |  ```
  HTML COMMENT CLOSING TAG    |  -->
  BLANK LINE                  |
  `````

</details>

<details>
<summary>

### Plaintext Comment Right-Align (4)

</summary>

`````markdown
  #### `win.flashFrame(flag)`   API HEADER
                                BLANK LINE
  <!--                          HTML COMMENT OPENING TAG
  ```YAML history               API HISTORY OPENING TAG
  added:                        API HISTORY
    - pr-url: https://...
  ```                           API HISTORY CLOSING TAG
  -->                           HTML COMMENT CLOSING TAG
                                BLANK LINE
  `````

</details>

Reply:
"Plaintext Comment Left-Align (3)" was ultimately chosen in Slack call.

Changed in [`f7d0095` (#42982)](https://github.com/electron/electron/pull/42982/commits/f7d00954749e53780569f15a995b0dc3e57fea77)